llvm/lib/Support/Unix/Threading.inc

```diff
@@ -137,13 +137,16 @@ uint64_t llvm::get_threadid() {
 }
 
 static constexpr uint32_t get_max_thread_name_length_impl() {
-#if defined(__NetBSD__)
+#if defined(PTHREAD_MAX_NAMELEN_NP)
   return PTHREAD_MAX_NAMELEN_NP;
 #elif defined(__APPLE__)
   return 64;
+#elif defined(__sun__) && defined(__svr4__)
+  return 31;
 #elif defined(__linux__) && HAVE_PTHREAD_SETNAME_NP
   return 16;
-#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) ||                   \
+    defined(__DragonFly__)
   return 16;
 #elif defined(__OpenBSD__)
   return 32;
```

Comment:
question (not part of this PR I know):
is it (still) true ? to me it looks like 24 (23+1).

ref [here](https://github.com/openbsd/src/blob/35183aa2ea28a86a3055f6daaea711b025aca5f9/sys/kern/kern_prot.c#L1159), [here](https://github.com/openbsd/src/blob/f8a2fc21e186f82477e86fa236a08c984f6ad143/sys/sys/syslimits.h#L83) and [here](https://github.com/openbsd/src/blob/f8a2fc21e186f82477e86fa236a08c984f6ad143/sys/sys/proc.h#L391).